### PR TITLE
Experimental mloadresolver

### PR DESCRIPTION
**Pull request by hrkrshnn**

### CODE REVIEW

File: test/libyul/yulOptimizerTests/fullSuite/proxy.yul [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+    let _1 := 0
+    calldatacopy(128, _1, calldatasize())
+    mstore(add(128, calldatasize()), _1)
+    pop(delegatecall(gas(), 25, 128, calldatasize(), _1, _1))
+    let data := _1
+    switch returndatasize()
+    case 0 { data := 96 }
+    default {
+        let result := and(add(returndatasize(), 63), not(31))
+        let memPtr := mload(64)
```
Comment (hrkrshnn) [inline test/libyul/yulOptimizerTests/fullSuite/proxy.yul:13]: Example from https://github.com/ethereum/solidity/issues/10645. This `mload(64)` can be resolved.

File: libyul/optimiser/DataFlowAnalyzer.cpp [hunk 2]
```diff
@@ -433,3 +482,222 @@ std::optional<YulString> DataFlowAnalyzer::isSimpleLoad(
...
+			arguments.at(0) / arguments.at(1)
+		);
+
+	// Restrictions from EIP-1985
+	case evmasm::Instruction::CALLDATASIZE:
+	case evmasm::Instruction::CODESIZE:
+	case evmasm::Instruction::EXTCODESIZE:
+	case evmasm::Instruction::MSIZE:
+	case evmasm::Instruction::RETURNDATASIZE:
+		return newRestrictedVariable(bigint(1) << 32);
```
Comment (chriseth) [inline libyul/optimiser/DataFlowAnalyzer.cpp:545]: I think `1<<64` is safer.

Comment (chriseth) [reply libyul/optimiser/DataFlowAnalyzer.cpp:545]: Ah sorry, it's the EIP!

File: libyul/optimiser/DataFlowAnalyzer.cpp [hunk 3]
```diff
@@ -433,3 +482,219 @@ std::optional<YulString> DataFlowAnalyzer::isSimpleLoad(
...
+		// Avoiding wrapping
+		m_solver->push();
+		m_solver->addAssertion(arguments.at(0) + arguments.at(1) >= (bigint(1) << 256));
+		CheckResult result = m_solver->check({}).first;
+		m_solver->pop();
+		if (result == CheckResult::UNSATISFIABLE)
+			return arguments.at(0) + arguments.at(1);
+		else
+			return newRestrictedVariable();
+	}
```
Comment (hrkrshnn) [inline libyul/optimiser/DataFlowAnalyzer.cpp:537]: @ekpyron here is an example of relaxation.

**Comment by hrkrshnn** [comment]
Closing this to reduced PR count. The PR has enough information for future needs.

**Pull request closed without merging**